distro/exporter_test.go

```diff
@@ -155,6 +156,158 @@ func TestOTLPProtocol(t *testing.T) {
+
+		assert.Equal(t, otlpProtocolHTTPProtobuf, got)
+		assert.Contains(t, buf.String(), fmt.Sprintf("invalid %s: %q", otelMetricsExporterOTLPProtocolKey, invalidProtocol))
+		assert.Contains(t, buf.String(), "falling back to")
+	})
+
+	// Logs
+	t.Run("logs: default", func(t *testing.T) {
+		buf, logger := newTestLogger()
+
+		got := otlpProtocol(logger, otelLogsExporterOTLPProtocolKey)
+
+		assert.Equal(t, defaultOTLPProtocol, got)
+		assert.Empty(t, buf.String())
+	})
+
+	t.Run("logs: only general protocol", func(t *testing.T) {
+		t.Setenv(otelExporterOTLPProtocolKey, otlpProtocolHTTPProtobuf)
+		buf, logger := newTestLogger()
+
+		got := otlpProtocol(logger, otelLogsExporterOTLPProtocolKey)
+
+		assert.Equal(t, otlpProtocolHTTPProtobuf, got)
+		assert.Empty(t, buf.String())
+	})
+
+	t.Run("logs: only specific protocol", func(t *testing.T) {
+		t.Setenv(otelLogsExporterOTLPProtocolKey, otlpProtocolHTTPProtobuf)
+		buf, logger := newTestLogger()
+
+		got := otlpProtocol(logger, otelLogsExporterOTLPProtocolKey)
+
+		assert.Equal(t, otlpProtocolHTTPProtobuf, got)
+		assert.Empty(t, buf.String())
+	})
+
+	t.Run("logs: specific overrides general", func(t *testing.T) {
+		t.Setenv(otelExporterOTLPProtocolKey, defaultOTLPProtocol)
+		t.Setenv(otelLogsExporterOTLPProtocolKey, otlpProtocolHTTPProtobuf)
+		buf, logger := newTestLogger()
+
+		got := otlpProtocol(logger, otelLogsExporterOTLPProtocolKey)
+
+		assert.Equal(t, otlpProtocolHTTPProtobuf, got)
+		assert.Empty(t, buf.String())
+	})
+
+	t.Run("logs: invalid specific value", func(t *testing.T) {
+		t.Setenv(otelLogsExporterOTLPProtocolKey, invalidProtocol)
+		buf, logger := newTestLogger()
+
+		got := otlpProtocol(logger, otelLogsExporterOTLPProtocolKey)
+
+		assert.Equal(t, defaultOTLPProtocol, got)
+		assert.Contains(t, buf.String(), fmt.Sprintf("invalid %s: %q", otelLogsExporterOTLPProtocolKey, invalidProtocol))
+		assert.Contains(t, buf.String(), "falling back to")
+		assert.Contains(t, buf.String(), otelExporterOTLPProtocolKey)
+	})
+
+	t.Run("logs: invalid general value", func(t *testing.T) {
+		t.Setenv(otelExporterOTLPProtocolKey, invalidProtocol)
+		buf, logger := newTestLogger()
+
+		got := otlpProtocol(logger, otelLogsExporterOTLPProtocolKey)
+
+		assert.Equal(t, defaultOTLPProtocol, got)
+		assert.Contains(t, buf.String(), fmt.Sprintf("invalid %s: %q", otelExporterOTLPProtocolKey, invalidProtocol))
+		assert.Contains(t, buf.String(), "using default")
+	})
+
+	t.Run("logs: invalid specific, valid general", func(t *testing.T) {
+		t.Setenv(otelLogsExporterOTLPProtocolKey, invalidProtocol)
+		t.Setenv(otelExporterOTLPProtocolKey, otlpProtocolHTTPProtobuf)
+		buf, logger := newTestLogger()
+
+		got := otlpProtocol(logger, otelLogsExporterOTLPProtocolKey)
+
+		assert.Equal(t, otlpProtocolHTTPProtobuf, got)
+		assert.Contains(t, buf.String(), fmt.Sprintf("invalid %s: %q", otelLogsExporterOTLPProtocolKey, invalidProtocol))
+		assert.Contains(t, buf.String(), "falling back to")
+	})
```

Comment:
I do not see any value in these tests. They basically test the same. I propose to revert changes from `exporter_test.go`.